x-pack/plugins/cases/public/components/all_cases/use_all_cases_state.tsx

```diff
@@ -0,0 +1,224 @@
+  const [filterOptions, setFilterOptions] = useState<FilterOptions>({
+    ...DEFAULT_FILTER_OPTIONS,
+    ...initialFilterOptions,
+  });
+
+  const [localStorageQueryParams, setLocalStorageQueryParams] =
+    useLocalStorage<LocalStorageQueryParams>(getQueryParamsLocalStorageKey(appId));
+
+  const [localStorageFilterOptions, setLocalStorageFilterOptions] =
+    useLocalStorage<PartialFilterOptions>(getFilterOptionsLocalStorageKey(appId));
+
+  const persistAndUpdateQueryParams = useCallback(
+    (params) => {
+      if (isModalView) {
+        setQueryParams((prevParams) => ({ ...prevParams, ...params }));
+        return;
+      }
+
+      const parsedUrlParams: ParsedUrlQueryParams = parse(location.search);
+      const urlParams: PartialQueryParams = parseUrlQueryParams(parsedUrlParams);
+      const newQueryParams: QueryParams = getQueryParams(
+        params,
+        queryParams,
+        urlParams,
+        localStorageQueryParams
+      );
+      const newLocalStorageQueryParams = {
+        perPage: newQueryParams.perPage,
+        sortField: newQueryParams.sortField,
+        sortOrder: newQueryParams.sortOrder,
+      };
+      setLocalStorageQueryParams(newLocalStorageQueryParams);
+      setQueryParams(newQueryParams);
+    },
+    [isModalView, location.search, queryParams, localStorageQueryParams, setLocalStorageQueryParams]
+  );
+
+  const persistAndUpdateFilterOptions = useCallback(
+    (params) => {
+      if (isModalView) {
+        setFilterOptions((prevParams) => ({ ...prevParams, ...params }));
+        return;
+      }
+
+      const newFilterOptions: FilterOptions = getFilterOptions(
+        filterOptions,
+        params,
+        parse(location.search),
+        localStorageFilterOptions
+      );
+
+      const newPersistedFilterOptions: PartialFilterOptions = {
+        ...(newFilterOptions.severity && { severity: newFilterOptions.severity }),
+        ...(newFilterOptions.status && { status: newFilterOptions.status }),
+      };
+
+      const newLocalStorageFilterOptions: PartialFilterOptions = {
+        ...localStorageFilterOptions,
+        ...newPersistedFilterOptions,
+      };
+      setLocalStorageFilterOptions(newLocalStorageFilterOptions);
+      setFilterOptions(newFilterOptions);
+    },
+    [
+      filterOptions,
+      isModalView,
+      localStorageFilterOptions,
+      location.search,
+      setLocalStorageFilterOptions,
+    ]
+  );
+
+  useEffect(() => {
+    if (isFirstRenderRef.current) {
+      persistAndUpdateQueryParams(isModalView ? queryParams : {});
+      persistAndUpdateFilterOptions(isModalView ? filterOptions : initialFilterOptions);
+
+      isFirstRenderRef.current = false;
+    }
+  }, [
+    filterOptions,
+    initialFilterOptions,
+    isModalView,
+    persistAndUpdateFilterOptions,
+    persistAndUpdateQueryParams,
+    queryParams,
+  ]);
+
+  if (!isModalView) {
+    const parsedUrlParams = parse(location.search);
```

Comment:
If invalid sorting params are supplied in the URL should we reset the invalid params to the defaults? For example entering a URL of

```
https://localhost:5601/abc/app/observability/cases?page=1&perPage=25&severity=all&sortField=titl&sortOrder=bsc&status=in-progress
```

An empty list is displayed with the error:

<img width="1640" alt="image" src="https://user-images.githubusercontent.com/7405507/211627623-ffc40245-9fb1-4e50-b178-bba5b23118ca.png">

Comparing this with the behavior of `perPage` for example, if I use `perPagx` the page loads correctly and adds the default of `perPage=25` into the URL.

Reply:
I had a short talk with @cnasikas and he reminded me that when we did the initial PR for adding parameters we decided not to handle this.

> Comparing this with the behavior of perPage for example, if I use perPagx the page loads correctly and adds the default of perPage=25 into the URL.

I thought this was weird because there is currently no validation for any of these parameters. I tried to reproduce it and what happens is that if you add `perPagex` that one is ignored and we leave it in the URL. `perPage` is added from your local storage so we end up with something like `perPagx=25&perPage=25`.

The same should already happen with misspelled `severity` or `status`.

What we don't look for is the values of these fields. So if `status=Oppen` e.g. the load will fail.

I think this scenario is not very likely though because we expect the users to use the UI on the page and not directly type their searches on the URL.

Maybe I can create an issue if we want to handle it in the future?

Reply:
Agreed that this is a fairly unlikely scenario, and the error toast that is displayed gives a good indication to the user what the problem is, so no need to raise a separate issue for this. [Adding the ability to clear filters from the page](https://github.com/elastic/kibana/issues/148884) should also make it easy for the user to clear the invalid parameter from the URL.